scripts/build_static_deps.sh

```diff
@@ -23,6 +23,14 @@ BUILD_DEST="${REPO_DIR}/build/lib"
 
 git submodule update --init
 
+if ${MAKE} -C "${ROCKSDB_LIB_DIR}" --dry-run static_lib | grep -q 'Nothing to be done'; then
+  echo "RocksDb static libraries already built. Skipping build."
+  exit 0
+else
+  ${REPO_DIR}/scripts/clean_build_artifacts.sh
+  echo "Building RocksDb static libraries."
+fi
```

Comment:
@arnetheduck This should fix the problem you encountered. If we have already built RocksDb then it won't be rebuilt unless you switch the code then it will check if a rebuild is needed, and will clean the build artifacts just before completing the full rebuild. 

By the way I couldn't find a good way to clean up the build artifacts as you suggested without triggering RocksDb to rebuild and we don't want to rebuild RocksDb every time, that would be too slow. This is because the make file targets are all linked together and if any files are changed or deleted then it will trigger a rebuild up the path of what was changed.

Reply:
alright! nice bonus with the echo, without it looks like the build has frozen..